src/metabase/models/interface.clj

```diff
@@ -178,21 +182,20 @@
     (post-select <>)
     (map-> entity <>)))
 
-
 (def IEntityDefaults
   "Default implementations for `IEntity` methods."
-  {:default-fields       (constantly nil)
-   :timestamped?         (constantly false)
-   :hydration-keys       (constantly nil)
-   :types                (constantly nil)
-   :can-read?            (constantly true)
-   :can-write?           superuser?
-   :pre-insert           identity
-   :post-insert          identity
-   :pre-update           identity
-   :post-update          (constantly nil)
-   :post-select          identity
-   :pre-cascade-delete   (constantly nil)})
+  {:default-fields            (constantly nil)
+   :timestamped?              (constantly false)
+   :hydration-keys            (constantly nil)
+   :types                     (constantly nil)
+   :can-read?                 (constantly true)
+   :can-write?                superuser?
```

Comment:
just looking at these defaults now and thinking that it may actually be better to use `(constantly false)` for the permissions defaults.  i know it's convenient to have these be the most likely values, but this is the kind of thing that becomes a security bug eventually, so I'd vote for closing these off by default and forcing each entity to explicitly set its read/write check.